### PR TITLE
feat: Search cancellation from Metastore with global context cache using Redis

### PR DESCRIPTION
**Pull request by sumandas0**
## Search cancellation from Metastore with global context cache using Redis

  > https://www.notion.so/atlanhq/Indexsearch-request-cancellation-from-the-server-25bffe2dad10407397b6c04768296b15?pvs=4

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

### CODE REVIEW

**Review by checkaayush — DISMISSED** [review]
LGTM

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java [hunk 1]
```diff
@@ -218,17 +240,21 @@ private DirectIndexQueryResult performAsyncDirectIndexQuery(SearchParams searchP
...
+            // Extract search context ID and sequence number
+            String currentSearchContextId = searchParams.getSearchContextId();
+            Integer currentSequenceNumber = searchParams.getSearchContextSequenceNo();
+            // Get the search ID from the cache if sequence number is greater than the current sequence number
+            String previousESSearchId = SearchContextCache.getESAsyncSearchIdFromContextCache(currentSearchContextId, currentSequenceNumber);
+
+            if (StringUtils.isNotEmpty(previousESSearchId)) {
+                LOG.debug("Deleting the previous async search response with ID {}", previousESSearchId);
+                // If the search ID exists, then we need to delete the search context
+                deleteAsyncSearchResponse(previousESSearchId);
```
Comment (checkaayush) [inline graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:254]: Should we handle exceptions thrown from `deleteAsyncSearchResponse` here?

**Review by checkaayush — APPROVED** [review]
Left a minor comment. Rest LGTM.

**Review by mehtaanshul — APPROVED** [review]

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java (hunk 1 above)
Comment (sumandas0) [reply graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:254]: Anyway this is async function and its already handled

Comment (sumandas0) [reply graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:254]: And parent function i already handled

**Pull request merged**